api_tests/resources/views/test_resource_detail.py

```diff
@@ -131,28 +166,346 @@ def test_anonymized_data(self, app):
+        test_artifact.finalized = False
+        test_artifact.save()
+
+        payload = make_patch_payload(test_artifact, is_finalized=True)
+        resp = app.patch_json_api(make_api_url(test_artifact), payload, auth=test_auth, expect_errors=True)
+
+        assert resp.status_code == 409
+        error_info = resp.json['errors'][0]
+        assert error_info['source']['pointer'] == expected_error_source
 
+    def test_patch_finalized__patch_is_atomic(self, app):
+        test_artifact, test_auth, _ = configure_test_preconditions()
+        test_artifact.artifact_type = ArtifactTypes.UNDEFINED
+        test_artifact.save()
+
+        original_title = test_artifact.title
+        original_description = test_artifact.description
+        original_identifier = test_artifact.identifier
+
+        payload = make_patch_payload(
+            test_artifact,
+            new_title='Some new name',
+            new_description='Some new desciprtion',
+            new_pid='Some new pid',
+            new_resource_type=ArtifactTypes.UNDEFINED,
+            is_finalized=True
+        )
+
+        app.patch_json_api(make_api_url(test_artifact), payload, auth=test_auth, expect_errors=True)
+        test_artifact.refresh_from_db()
+
+        assert test_artifact.title == original_title
+        assert test_artifact.description == original_description
+        assert test_artifact.artifact_type == ArtifactTypes.UNDEFINED
+        assert test_artifact.identifier == original_identifier
+        assert test_artifact.identifier.value == original_identifier.value
+        assert not test_artifact.finalized
+
+    def test_patch_finalized__adds_log(self, app):
+        test_artifact, test_auth, registration = configure_test_preconditions()
+        assert not registration.logs.filter(action=NodeLog.RESOURCE_ADDED).exists()
+
+        test_artifact.finalized = False
+        test_artifact.save()
+
+        payload = make_patch_payload(test_artifact, is_finalized=True)
+        app.patch_json_api(make_api_url(test_artifact), payload, auth=test_auth, expect_errors=True)
+
+        resource_added_log = registration.logs.get(action=NodeLog.RESOURCE_ADDED)
+        assert resource_added_log.params['artifact_id'] == test_artifact._id
+        assert resource_added_log.params['new_identifier'] == test_artifact.identifier.value
+
+
+@pytest.mark.django_db
+class TestResourceDetailDELETEPermissions:
+
+    @pytest.mark.parametrize('registration_state', [RegStates.ACCEPTED, RegStates.EMBARGO])
+    def test_status_code__admin(self, app, registration_state):
+        test_artifact, test_auth, _ = configure_test_preconditions(
+            registration_state=registration_state, user_role=UserRoles.ADMIN_USER
+        )
+        resp = app.delete_json_api(make_api_url(test_artifact), auth=test_auth, expect_errors=True)
+        assert resp.status_code == 204
+
+    @pytest.mark.parametrize('user_role', UserRoles.excluding(UserRoles.ADMIN_USER))
```

Comment:
So Write users can add and edit, but only Admins can delete? That seems odd.

Reply:
Wrote this before we agreed that WRITE users should be able to POST, in which case WRITE users should absolutely be able to DELETE. Good catch, thanks!